QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,199 @@
+The features are tested on module(unit) -integration and system test level. [Guidelines](https://iceoryx.io/latest/advanced/best-practice-for-testing/) for Contributors ensure a high quality for test development.
+
+### Coverage [4.iii]
+
+The Eclipse iceoryx project use [gcov](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html) to capture the following coverage types:
+
+- Line Coverage
+- Function Coverage
+- Branch Coverage
+
+The coverage results of every Pull-Request and master branch are publicly available in [codecov.io](https://app.codecov.io/gh/eclipse-iceoryx/iceoryx).
+If the user want a more detailed report (e.g. the coverage in the different packages), he can follow [this](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coverage-scan) guide to obtain them locally.
+
+### Performance [4.iv]
+
+The most important measurement units for performance testing on iceoryx are the data transfer latency and the CPU load.
+They can be obtained in two ways:
+
+1. [iceperf](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_examples/iceperf#iceperf)
+This performance test uses directly the iceoryx API without any indirections to compare different IPC mechanisms on increasing payload sizes. The output is a table linking the payload sizes with the latency in microseconds.
+
+2. [performance_test](https://gitlab.com/ApexAI/performance_test/-/tree/master/)
+The `performance_test` package tests various middleware implementations in ROS using inter-process communication.
+It offers a communication binding for iceoryx and gives information about latency, CPU usage etc.
+Additionally it is possible to test here the integration of iceoryx + CycloneDDS in ROS 2 using the rmw communication abstraction.
+
+Currently there are no automatic regression tests in the CI running, the work to integrate this is addressed in [https://github.com/eclipse-iceoryx/iceoryx/issues/137](https://github.com/eclipse-iceoryx/iceoryx/issues/137). The performance tests are currently executed manually before every release that will integrated into ROS by the maintainers to avoid performance regression.
+
+### Linters and Static Analysis [4.v]
+
+The code formatting is enforced through the CI using `clang-format` and can be used by following this [guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style)
+
+The code base is linted with `clang-tidy` to enforce naming conventions and basic coding rules and checked in the Pull-Requests by review.
+A [static code analysis](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#static-code-analysis) is done regularly with the Axivion Suite to check against the the [AUTOSAR C++14](https://www.autosar.org/fileadmin/user_upload/standards/adaptive/17-03/AUTOSAR_RS_CPP14Guidelines.pdf) rules, but the results are not publicly accessible due to licensing.
```

Comment:
Instead of linking to the AUTOSAR rules directly, you could also link to the configuration file of Axivion here. IIRC it also contains few MISRA and CERT rules.